app/src/main/java/net/gsantner/opoc/ui/SearchOrCustomTextDialog.java

```diff
@@ -268,7 +268,10 @@ public static void showMultiChoiceDialogWithSearchFilterUI(final Activity activi
         // =========================================================================================
 
         // Ok button only present under these circumstances
-        if (((dopt.callback != null && dopt.isSearchEnabled) || (dopt.positionCallback != null && dopt.isMultiSelectEnabled))) {
+        final boolean isSearchOk = dopt.callback != null && dopt.isSearchEnabled;
+        final boolean isMultiSelOk = dopt.positionCallback != null && dopt.isMultiSelectEnabled;
+        final boolean isPlainDialog = dopt.callback != null && (dopt.data == null || dopt.data.size() == 0);
```

Comment:
similar to the filemanager button calculations. very good improvement and to understand when reading it